### PR TITLE
EEP: Moving Forward Together with Modern Java

### PR DESCRIPTION
**Pull request by dmikurube**
This is another "retrospective" EEP to explain what we have done for modern Java support, and what we would do in the future.

Read the rendered version at:

https://github.com/embulk/embulk/blob/eep-modern-java/docs/eeps/draft-modern-java.md

### CODE REVIEW

File: docs/eeps/draft-modern-java.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+In light of the foregoing, we will consider loading the Embulk core as a "named module" when the following conditions are satisfied.
+
+* After Embulk removes `msgpack-core` from `embulk-spi`.
+* After Embulk unsupports Java 8.
+* After Embulk makes some decision about the logging driver.
+
+In other words, this will happen after Embulk v2.0 at the earliest. See also [EEP-8: Milestones to Embulk v1.0, and versioning strategies to follow](./eep-0008.md).
+
+However, even before the transition happens and the JAR files start having `module-info.class`, we may anyway want to start adding [the `Automatic-Module-Name` manifest attribute earlier in `META-INF/MANIFEST.MF`](https://docs.oracle.com/en/java/javase/11/docs/specs/jar/jar.html#modular-jar-files) to every JAR file, including `embulk-spi`, `embulk-core`, `embulk-deps`, many other `embulk-util-*` libraries, and plugins. At the same time, we'll have to take care of the merge of `embulk-spi` into `embulk-core` when adding the `Automatic-Module-Name` attribute.
```
Comment (hiroyuki-sato) [inline docs/eeps/draft-modern-java.md:83]: > we may anyway want to start adding [the `Automatic-Module-Name` manifest attribute earlier in `META-INF/MANIFEST.MF`](https://docs.oracle.com/en/java/javase/11/docs/specs/jar/jar.html#modular-jar-files) to every JAR file, including `embulk-spi`, `embulk-core`, `embulk-deps`, many other `embulk-util-*` libraries, and plugins.

Will this part work before removing `msgpack-core` from `embulk-spi`?

It means

* Adding Automatic-Module-Name
* After Embulk removes `msgpack-core` from `embulk-spi`.
* After Embulk unsupports Java 8.
* After Embulk makes some decision about the logging driver.

**Review by hiroyuki-sato — APPROVED** [review]
LGTM👍 Sorry for reviewing late. (I left one questions Just for interest)

File: docs/eeps/draft-modern-java.md (hunk 1 above)
Comment (dmikurube) [reply docs/eeps/draft-modern-java.md:83]: Sorry, I didn't get the point of your question... At least I meant that "adding Automatic-Module-Name" would happen before removing `msgpack-core`, as written as "start adding the `Automatic-Module-Name` manifest attribute **earlier**".

**Comment by dmikurube** [comment]
> Sorry for reviewing late.

No problems! Thanks for reviewing and approving.

**Comment by dmikurube** [comment]
Let me merge this -- let's keep on the comment if you have more questions. :)

**Pull request merged**